addons/account_payment/static/src/js/payment_form.js

```diff
@@ -0,0 +1,22 @@
+/** @odoo-module **/
+
+import PaymentForm from '@payment/js/payment_form';
+PaymentForm.include({
+    _prepareTransactionRouteParams() {
+        let transactionRouteParams = this._super(...arguments);
+        let postParams = JSON.parse(pythonDictToJSON(this.paymentContext['postParams']));
+        transactionRouteParams = { ...transactionRouteParams, ...postParams };
+        return transactionRouteParams;
+    },
+});
+
+function pythonDictToJSON(pythonDict) {
```

Comment:
not really fan of this, i think we can find a way to avoid doing a custom function for this

Reply:
I can do it without a custom function but it won't be so clear, what options do u think of?